### ISSUE TITLE
mergeStyles: Switching html direction to rtl doesn't change style direction

### ISSUE THREAD

**Issue by jordandrako** [OfficeDev/office-ui-fabric-react#3903]
### Bug Report

- __Package version(s)__: latest

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

#### Describe the issue:

When switching the HTML direction to RTL, the styles (such as padding-left, left, etc) aren't changing directions (to padding-right, right, etc) like the RTL sass mixin did.

Here's a screenshot of the HTML changed to RTL, using the searchBox example:
![image](https://i.imgur.com/CRJpZrs.jpg)

#### Actual behavior:

Direction related styles remain unaffected after switching the HTML direction.

#### Expected behavior:

Direction related styles should re-render to the correct direction.

**Comment by dzearing** [comment]
This should be fixed in my last push. The `dir` attribute was being written to the document late. I've adjusted the logic slightly to ensure reading the dir propagates to mergeStyles.

**Comment by dzearing** [comment]
@jordandrako can you verify this is better now and close, or ping me if not?

**Comment by jordandrako** [comment]
@dzearing I pulled from master and I'm still seeing padding-left on the SearchBox root element.

**Comment by fpintos** [comment]
I've hit the same issue. From my debugging, the cause is that rtlifyRules is not being called during mergeStyleSets because styleToRegistration finds a cached item and uses it, without going thru serializeRuleEntries.

Perhaps getKeyForRules should include the state of RTL in it, so when it changes, we will get a different key and serialize the new style.

**Comment by jordandrako** [comment]
@dzearing This is still an issue. I'm hesitant to push through any more mergeStyles conversions on high traffic components until this is resolved.
